src/views/Correction/FmCorrection.vue

```diff
@@ -114,6 +115,10 @@ export default class FmCorrection extends Mixins(CommonMixin, DateMixin, FilingT
       // safety check
       if (!this.correctionFiling) throw (new Error('Missing correction filing'))
 
+      // fetch and store business snapshot
+      const businessSnapshot = await this.fetchBusinessSnapshot()
+      this.parseEntitySnapshot(businessSnapshot)
+
```

Comment:
I moved this as it was overwritting the changes that parse correction filing was setting

Reply:
The business snapshot and the draft correction filing are separate pieces of data (original and changed, respectively). They should not clobber each other, and in fact there are many methods that compare original vs changed to identify if something has changed.

I am actually working on similar code for BEN IA corrections and will look at this also after your PR is merged.

Reply:
I'm not concerned about this code in this PR since firm corrections are currently incomplete, but I think this should be a call to `setEntitySnapshot()` not parseEntitySnapshot().

The idea is that the business snapshot is saved in one place and the "current data" (being corrected) is saved separately (in various store objects according to the components used and how the code was developed originally).

Reply:
Um, OK, parseEntitySnapshot() calls setEntitySnapshot(), but it also sets other data in the store, possibly incorrectly. I'm looking into this for BEN IA corrections and firm corrections so don't worry about this for now.